### PR TITLE
chore: update readme based on latest template

### PR DESCRIPTION
**Pull request by toddbaert**
This PR:

- updates the README based on latest template for automatic inclusion in http://openfeature.dev/

I've tried to get everything right, but there's 4 code-example TODOs. Maybe @jamescarr or @federicobond wouldn't mind filling in those gaps.

@beeme1mr feel free to make any corrections you see.

I also moved the file, but did that in a separate commit. Please see the first commit to see the diff.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+  </picture>
+</p>
+
+<h2 align="center">OpenFeature Python SDK</h2>
+
+<!-- x-hide-in-docs-end -->
+<!-- The 'github-badges' class is used in the docs -->
+<p align="center" class="github-badges">
+
+  <a href="https://github.com/open-feature/spec/tree/v0.3.0">
```
Comment (beeme1mr) [inline README.md:16]:
```suggestion
  <a href="https://github.com/open-feature/spec/releases/tag/v0.3.0">
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    <img alt="Specification" src="https://img.shields.io/static/v1?label=Specification&message=v0.3.0&color=red&style=for-the-badge" />
+  </a>
+
+  <!-- x-release-please-start-version -->
+
+  <a href="https://badge.fury.io/py/openfeature-sdk">
```
Comment (beeme1mr) [inline README.md:22]:
```suggestion
  <a href="https://github.com/open-feature/python-sdk/releases/tag/v0.3.1">
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+
+### Requirements
+
+- Python 3.8+
+
+### Install
+
+<!---x-release-please-start-version-->
+
+Pip install
```
Comment (beeme1mr) [inline README.md:60]:
```suggestion
#### Pip install
```

File: README.md [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+```bash
+pip install openfeature-sdk==0.3.1
+```
+
+requirements.txt
```
Comment (beeme1mr) [inline README.md:66]:
```suggestion
#### requirements.txt
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+```
+
+#### requirements.txt
+
+```bash
+openfeature-sdk==0.3.1
+```
+
+```python
+pip install requirements.txt
```
Comment (federicobond) [inline README.md:73]:
```suggestion
pip install -r requirements.txt
```

File: README.md [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+
+global_context = EvaluationContext(
+    targeting_key="targeting_key1", attributes={"application": "value1"}
+)
+request_context = EvaluationContext(
+    targeting_key="targeting_key2", attributes={"email": request.form['email']}
+)
+
+## set global context
+set_evaluation_context(first_context)
```
Comment (federicobond) [inline README.md:143]:
```suggestion
set_evaluation_context(global_context)
```

**Review by federicobond — APPROVED** [review]

File: README.md [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+# set global hooks at the API-level
+from openfeature.api import add_hooks
+add_hooks([MyHook()])
+
+# or configure them in the client
+client = OpenFeatureClient()
+client.add_hooks([MyHook()])
+```
+
+<!-- TODO: example of invocation-level hook -->
```
Comment (federicobond) [inline README.md:169]:
```python
from openfeature.flag_evaluation import FlagEvaluationOptions

options = FlagEvaluationOptions(hooks=[MyHook()])
client.get_boolean_flag("my-flag", False, flag_evaluation_options=options)
```

File: README.md [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+
+A shutdown method is not yet available in the Python SDK. Progress on this feature can be tracked [here](https://github.com/open-feature/python-sdk/issues/125).
+
+## Extending
+
+### Develop a provider
+
+To develop a provider, you need to create a new project and include the OpenFeature SDK as a dependency.
+This can be a new repository or included in [the existing contrib repository](https://github.com/open-feature/python-sdk-contrib) available under the OpenFeature organization.
+You’ll then need to write the provider by implementing the `FeatureProvider` interface exported by the OpenFeature SDK.
```
Comment (federicobond) [inline README.md:193]: It's called `AbstractProvider` in the SDK, though I have been wondering if we should rename it for consistency with other SDKs.

File: README.md [hunk 9]
```diff
@@ -0,0 +1,241 @@
...
+
+<!-- TODO: code example of provider implementation, see: https://github.com/open-feature/java-sdk#usage -->
```
Comment (federicobond) [inline README.md:195]:
```python
from typing import List, Optional

from openfeature.evaluation_context import EvaluationContext
from openfeature.flag_evaluation import FlagResolutionDetails
from openfeature.provider.provider import AbstractProvider

class MyProvider(AbstractProvider):
    def get_metadata(self) -> Metadata:
        ...

    def get_provider_hooks(self) -> List[Hook]:
        return []

    def resolve_boolean_details(
        self,
        flag_key: str,
        default_value: bool,
        evaluation_context: Optional[EvaluationContext] = None,
    ) -> FlagResolutionDetails[bool]:
        ...

    def resolve_string_details(
        self,
        flag_key: str,
        default_value: str,
        evaluation_context: Optional[EvaluationContext] = None,
    ) -> FlagResolutionDetails[str]:
        ...

    def resolve_integer_details(
        self,
        flag_key: str,
        default_value: int,
        evaluation_context: Optional[EvaluationContext] = None,
    ) -> FlagResolutionDetails[int]:
        ...

    def resolve_float_details(
        self,
        flag_key: str,
        default_value: float,
        evaluation_context: Optional[EvaluationContext] = None,
    ) -> FlagResolutionDetails[float]:
        ...

    def resolve_object_details(
        self,
        flag_key: str,
        default_value: Union[dict, list],
        evaluation_context: Optional[EvaluationContext] = None,
    ) -> FlagResolutionDetails[Union[dict, list]]:
        ...
```

File: README.md (hunk 8 above)
Comment (toddbaert) [reply README.md:193]: I'm not big on enforcing specific class/method/function/member names. I think that's something that the implementation should have a lot of flexibility on.

For example, I understand there isn't formal `Interfaces` in Python, due to multiple inheritance etc... so if an abstract base class is the right choice, and `Abstract` is a common prefix for such classes, then `AbstractProvider` makes perfect sense. **_I think the important thing to consider is what naming will communicate the core _ideas_ of the SDK to a Python developer most clearly_** - I trust your ability to answer that question much more than mine.

File: README.md [hunk 10]
```diff
@@ -0,0 +1,310 @@
...
+### Usage
+
+```python
+from openfeature import api
+from openfeature.provider.in_memory_provider import InMemoryFlag, InMemoryProvider
+
+# flags defined in memory
+my_flags = {
+  "v2_enabled": InMemoryFlag("on", {"on": True, "off": False})
+}
```
Comment (beeme1mr) [inline README.md:87]: This configuration requires [this change](https://github.com/open-feature/python-sdk/pull/228) to be released.

File: README.md (hunk 8 above)
Comment (beeme1mr) [reply README.md]: I've updated the comment based on everyones feedback.

**Review by beeme1mr — APPROVED** [review]

File: README.md (hunk 10 above)
Comment (beeme1mr) [reply README.md:87]: I'll release `0.4.0` right after merging this.

**Pull request merged**